src/actions/setCursor.ts

```diff
@@ -163,7 +164,12 @@ const setCursor = (
 /** Action-creator for setCursor. */
 export const setCursorActionCreator =
   (payload: Parameters<typeof setCursor>[1]): Thunk =>
-  dispatch =>
+  dispatch => {
+    // If editing is being set to true, close the command menu first
+    if (payload.editing === true) {
+      dispatch(toggleCommandMenuActionCreator({ value: false }))
```

Comment:
It would be preferable to push this all the way down to the reducer. Action-creators are better for browser state and side effects, while reducers are better for pure input/output.

Followup on https://github.com/cybersemics/em/pull/2874#discussion_r2019971281.